### DISCUSSION TITLE
adding child when adding parent

### DISCUSSION THREAD

**Discussion by MattRvt** [payloadcms/payload#638] [Q&A] [answered]
Is this possible to add children when receiving post request with this kind of content:
http://localhost:3000/api/parent?depth=1
``` JSON
{
    "name": "parent",
    "child":{
        "name": "child"
    }
}
```
This would create child and parent object. Instead, I get ```child: null``` in the answer

**Comment by jmikrut** [comment, marked as answer]
Hey @MattRvt — right now, this is not possible. It _**could**_ be a a future feature that we build, but it'll take some new functionality to achieve. The way we'd do this currently is to first create the nested document, then pass the newly created document's ID to your parent doc.

So:

```js
const parentDocData = {
    "name": "parent",
    "child": "6a3434543qw3derra" // ID of newly created child doc
}
```

**Reply by MattRvt** [reply]
Thank you for your reply. This feature would be awesome.

The same way, being able to create child through the UI from the parent page as it is done with array would be a great feature.

Again, many thanks for your work, your reply and the great documentation 🥳

**Reply by jmikrut** [reply]
You got it!

Actually we are definitely going to allow creating a child through the UI from the parent page. That's on the radar. Stay tuned!
